Update WooCommerce developer documentation

rodrigoprimo opened the pull request:
### Changes proposed in this Pull Request:

This PR updates the following files: README.md, CONTRIBUTING.md, and tests/README.md. The overall goal of these changes is to make it easy for new WooCommerce contributors to find information about how to get started.

Here is a detailed list of changes per file:

CONTRIBUTING.md:

- Section about translating WooCommerce moved to its [own wiki page](https://github.com/woocommerce/woocommerce/wiki/Translating-WooCommerce)
- Section about the build process consolidated in the wiki page about [how to set up a development environment](https://github.com/woocommerce/woocommerce/wiki/How-to-set-up-WooCommerce-development-environment)
- Update the list of ways to contribute to the project
- Add a sentence about the project license and another about the #core channel on Slack
- Add a section called "Getting started" with links to the primary documents for those wishing to contribute code to WooCommerce
- Update section with guidelines to include an item about fixing pre-existing code standards violations in the modified files and another item about good commit messages

README.md:

- Add a link pointing to the wiki
- Consolidate section about support with information moved from CONTRIBUTING.md

tests/README.md:

- Use composer instead of installing PHPUnit manually and some other minor changes

rodrigoprimo committed:
Update tests documentation

Update tests documentation as PHPUnit now should be installed and called using composer instead of
globally. This commit also includes some other minor changes to the
tests documentation like updating the link of the service used for code
coverage reports.

rodrigoprimo committed:
Consolidate info about support on the README.md file

This commit removes duplicated information about support from
CONTRIBUTING.md and consolidates it in README.md.

rodrigoprimo committed:
Add a link to the wiki in the README.md file

rodrigoprimo committed:
Move Translating WooCommerce section to its own page

https://github.com/woocommerce/woocommerce/wiki/Translating-WooCommerce

rodrigoprimo committed:
Remove section about the build process

This is already covered in the document about how to set up a
development enviroment.

rodrigoprimo committed:
Reorganize CONTRIBUTING.md

claudiosanches commented on .github/CONTRIBUTING.md:
```diff
@@ -1,112 +1,51 @@
+and [sendind a pull request](https://help.github.com/articles/using-pull-requests/), please see the information in the sections below.
 
-The source code found on GitHub does not contain compiled CSS or Javascript, and is missing some functionality such as Gutenberg Blocks and the REST API, until the build process is ran.
+WooCommerce is licensed under the GPLv2+, and all contributions to the project will be released under the same license. You maintain copyright over any contribution you make, and by submitting a pull request, you are agreeing to release that contribution under the GPLv2+ license.
```
Our license.txt says GPLv3+

rodrigoprimo replied:
I just pushed a commit to fix that. Thanks

claudiosanches reviewed (changes requested):
Looks good, just a minor issue.

haszari commented on .github/CONTRIBUTING.md:
```diff
@@ -1,112 +1,51 @@
 
-## Build process 🛠
+If you wish to contribute code, before [forking](https://help.github.com/articles/fork-a-repo/) WooCommerce, commiting your changes,
+and [sendind a pull request](https://help.github.com/articles/using-pull-requests/), please see the information in the sections below.
```
Typo: `sending` :) or maybe say `submitting a pull request`. Or maybe switch the sentence around, e.g. something like

`If you wish to contribute code, please read the information in the sections below. Then fork WooCommerce, commit your changes, and submit a pull request 🎉`

rodrigoprimo replied:
Thanks, @haszari. I liked your suggestion to change the sentence and pushed a commit to incorporate.

haszari reviewed (approved):
Awesome, thanks for moving this forward @rodrigoprimo ! I especially like the clear links to the key sections of the wiki, and the prompt to fix violations in existing files :)

I added one comment, just a suggestion really.

peterfabian commented on .github/CONTRIBUTING.md:
```diff
@@ -1,112 +1,51 @@
+- [Minification of SCSS and JS](Minification-of-SCSS-and-JS)
+- [String localisation guidelines](String-localisation-guidelines)
+- [Running unit tests](https://github.com/woocommerce/woocommerce/blob/master/tests/README.md)
+- [Running e2e tests](End-to-end-Testing)
```
Could be worth mentioning also our approach to GitHub branches, if to use merge or rebase to keep the branch up to date. What do you reckon?

rodrigoprimo replied:
I added a link to the Git Flow wiki page. This should cover our approach to branches. Regarding merging or rebasing, I don't have a strong opinion. I usually merge instead of rebasing. Maybe we should discuss this with the rest of the team and then update the Git Flow wiki page?

peterfabian replied:
I agree with the approach.

haszari replied:
Rebasing requires more instructions/guardrails, so recommending merging should be easier to wrangle :)

peterfabian reviewed (commented):
Great restructuring. Just have some small comments.

rodrigoprimo committed:
Fix GPL version

rodrigoprimo committed:
Improve paragraph about the process to contribute code to WC core

rodrigoprimo commented on .github/CONTRIBUTING.md:
```diff
@@ -1,112 +1,51 @@
-- Answering questions on GitHub and within the various WooCommerce communities.
-- Submitting fixes, improvements, and enhancements.
-
 WooCommerce currently powers 30% of all online stores across the internet, and your help making it even more awesome will be greatly appreciated :)
```
I don't know what is the source of this. This sentence was already part of the document before I started making modifications so I opted to leave it.

rodrigoprimo replied:
Checking `git blame`, it seems that this sentence was added by @mikejolley in ce17349792 so he might know more about it.

peterfabian replied:
Not really important, just something that caught my eye :)

claudiosanches replied:
Comes from BuiltWith, still those stats are different now: https://trends.builtwith.com/shop/traffic/Entire-Internet
I vote to change this string to:

> WooCommerce powers several online stores across the internet, and your help making it even more awesome will be greatly appreciated :)

Sounds better and there's no issues with sources and no need to update numbers :smile:

mikejolley replied:
@rodrigoprimo ¯\_(ツ)_/¯

rodrigoprimo replied:
I changed the text to incorporate @claudiosanches' suggestion.

rodrigoprimo committed:
Add link to the WooCommerce Git Flow page in the "Getting started" section

rodrigoprimo committed:
Fix links that point to the internal wiki

rodrigoprimo committed:
Add sentence about the `help wanted` label

rodrigoprimo commented on .github/CONTRIBUTING.md:
```diff
@@ -1,112 +1,51 @@
 
-## Build process 🛠
+If you wish to contribute code, before [forking](https://help.github.com/articles/fork-a-repo/) WooCommerce, commiting your changes,
+and [sendind a pull request](https://help.github.com/articles/using-pull-requests/), please see the information in the sections below.
```
I added a new sentence mentioning the `help wanted` label. I don't think we should expand too much on the general process to contribute code to keep the document simple and straightforward. But I definitely agree that this part is too compressed for a beginner contributor that is why we have the links to the GitHub documentation where people can learn more about this process. Maybe we can find another document that provides a good overview of the whole process for beginners and include it here? What do you think?

peterfabian replied:
Yup, maybe we can even write a new document, sth like "How to make your first PR". You're right it's probably not a good idea to include more details in this document.

claudiosanches reviewed (changes requested):
One last change to improve the description.

rodrigoprimo committed:
Remove WooCommerce market share from the CONTRIBUTING.md file

rodrigoprimo commented:
I believe I addressed all the comments. I am marking this for review again.

mikejolley commented:
@rodrigoprimo "several", so humble :) I'd change that to "many" tbh.

> several: more than two but not many.

rodrigoprimo committed:
Replace "several" with "many" in the first sentence

rodrigoprimo commented:
> @rodrigoprimo "several", so humble :) I'd change that to "many" tbh.

@mikejolley I actually had the same thought when I read "several" but couldn't think of a better word. I just pushed a new commit replacing "several" with "many". Thanks.

vedanshujain commented on .github/CONTRIBUTING.md:
```diff
@@ -1,112 +1,53 @@
 - Push the changes to your fork and submit a pull request on the master branch of the WooCommerce repository.
-
-Please avoid modifying the change-log directly or updating the .pot files. These will be updated by the WooCommerce team.
+- Make sure to write good and detailed commit messages (see [this post](https://chris.beams.io/posts/git-commit/) for more on this) and follow all the applicable sections of the pull request template.
```
I wonder if we should link [WordPress commit message guidelines](https://make.wordpress.org/core/handbook/best-practices/commit-messages/) instead of Chris Beams post.

rodrigoprimo replied:
I found Chris Beams post on a quick search, I don't mind changing it to something else. IMHO, we shouldn't change to the WordPress commit message guidelines as it includes some details that are specific to the WP workflow and don't apply to the WooCommerce workflow (for example, include the component or focus name in the commit message summary and instructions that apply only to Trac like using square brackets to reference changeset).

vedanshujain commented on README.md:
```diff
@@ -25,8 +26,11 @@ To disclose a security issue to our team, [please submit a report via HackerOne
+* If you have a problem, you may want to start with the [self help guide](https://docs.woocommerce.com/document/woocommerce-self-service-guide/).
+* The [WooCommerce.com premium support portal](https://woocommerce.com/contact-us/ ) for customers who have purchased themes or extensions.
 * [Our community forum on wp.org](https://wordpress.org/support/plugin/woocommerce) which is available for all WooCommerce users.
+* The WooCommerce Help and Share Facebook group.
```
Maybe add link (if there is one)

rodrigoprimo replied:
I don't have a link to this group. This was already in the documentation and I just moved it here. We might want to reevaluate this list.

vedanshujain commented on tests/README.md:
```diff
@@ -2,46 +2,43 @@
 
 ## Running Tests
 
 Simply change to the plugin root directory and type:
```
Maybe get rid of the word `Simply` as well? It can have a negative implication for the reader if they don't know how to do it.

rodrigoprimo replied:
Good point, I just pushed a commit removing the word "simply". Thanks.

vedanshujain reviewed (approved):
Left a few comments, which are in no way a blocker. This doc looks pretty good ❤️

rodrigoprimo committed:
Remove word "simply" from documentation

It can have a negative implication for the reader if they don't know how to do it
